plugins/discovery-multicast/src/main/java/org/elasticsearch/plugin/discovery/multicast/MulticastDiscoveryPlugin.java

```diff
@@ -44,8 +46,9 @@ public String name() {
     public String description() {
         return "Multicast Discovery Plugin";
     }
-    
+
     public void onModule(DiscoveryModule module) {
+        deprecationLogger.deprecated("[discovery-multicast] plugin will be removed in 5.0.0. Use unicast or any cloud discovery plugin.");
```

Comment:
Nit: In other instances, we say "will be removed in the next major version" but if nobody else raises an objection, I am also fine with this wording.